business/workloads.go

```diff
@@ -1718,11 +1718,15 @@ func fetchWorkload(ctx context.Context, layer *Layer, criteria WorkloadCriteria)
 		// Add the Proxy Status to the workload
 		for _, pod := range w.Pods {
 			if pod.HasIstioSidecar() {
-				ps, err := layer.ProxyStatus.GetPodProxyStatus(criteria.Namespace, pod.Name)
-				if err != nil {
-					log.Warningf("GetPodProxyStatus is failing for [namespace: %s] [pod: %s]: %s ", criteria.Namespace, pod.Name, err.Error())
+				if config.Get().ExternalServices.Istio.IstioAPIEnabled {
+					ps, err := layer.ProxyStatus.GetPodProxyStatus(criteria.Namespace, pod.Name)
+					if err != nil {
+						log.Warningf("GetPodProxyStatus is failing for [namespace: %s] [pod: %s]: %s ", criteria.Namespace, pod.Name, err.Error())
+					}
+					pod.ProxyStatus = castProxyStatus(ps)
+				} else {
+					pod.ProxyStatus = &models.ProxyStatus{}
```

Comment:
Having an empty proxy status will cause workloads to show a warning on their health details:

![Screenshot from 2022-12-15 09-47-30](https://user-images.githubusercontent.com/6226732/207936836-5d32849b-a9e4-433c-be6d-3165bcd54ad0.png)

This is more to do with how the frontend is calculating the workload health. I'm not sure we should show the workload as being unhealthy or show a warning when the istiod access has been disabled.

Reply:
@josunect I still don't think we should show a warning label here for the empty proxy status when istiod access is disabled. As it is, pods will always be in a "warning" state but if you've purposefully disabled registry access then imo the frontend shouldn't try to use proxy status for the health calculation if it has no way of ever getting it.